### PR TITLE
Tpetra: update modules loaded on crusher

### PR DESCRIPTION
**Pull request by jhux2**
@trilinos/tpetra
## Stakeholder Feedback
Improves Tpetra performance dashboard

## Testing
tested locally on ORNL/crusher

### CODE REVIEW

**Review by brian-kelley — APPROVED** [review]

**Comment by jhux2** [comment]
downstream build problems, pausing this

**Review by GrahamBenHarper — APPROVED** [review]

**Review by GrahamBenHarper — APPROVED** [review]

**Review by GrahamBenHarper — APPROVED** [review]

**Comment by GrahamBenHarper** [comment]
It's still blocked by that one Epetra test unfortunately
```
/vscratch1/trilinos/jaas/workspace/Trilinos_PR_cuda-10.1.243/Trilinos/packages/thyra/adapters/epetra/test/EpetraOperatorWrapper/EpetraOperatorWrapper_UnitTests.cpp:54:10: fatal error: Trilinos_Util_CrsMatrixGallery.h: No such file or directory
 #include "Trilinos_Util_CrsMatrixGallery.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

File: packages/tpetra/scripts/PerformanceTesting/crusher/configureBuild.sh
Comment (lucbv) [inline packages/tpetra/scripts/PerformanceTesting/crusher/configureBuild.sh:65]: I am not a big fan of `-L/${CRAY_BINUTILS_VERSION}/cce/x86_64/lib` but for some reason Cray does not provide a variable that points to the toolchain libraries : (

**Review by GrahamBenHarper — DISMISSED** [review]

**Comment by bartlettroscoe** [comment]
NOTE: I removed `AT: RETEST` and please don't set again until issue #10842 is resolved (which we now know what is causing this and will formulate a solution and get it merged to 'develop' which will fix your PR builds the next time they run).

**Comment by jhux2** [comment]
#10842 and #10836 are blockers.
Setting to WIP until both are fixed.

**Comment by bartlettroscoe** [comment]
FYI: The [above](https://github.com/trilinos/Trilinos/pull/10775#issuecomment-1213463411) PR testing iteration that just completed showed a new failing test [TrilinosInstallTests_find_package_Trilinos](https://trilinos-cdash.sandia.gov/test/10050314).  I just created the new issue #10865 to get to the bottom of this.

**Comment by bartlettroscoe** [comment]
@jhux2, if you merge the branch [10865-ats2-disable-shylu](https://github.com/trilinos/Trilinos/tree/10865-ats2-disable-shylu) to this topic branch [fix-crusher-modules](https://github.com/trilinos/Trilinos/tree/fix-crusher-modules) for this PR, then that changes should fix the test failure [TrilinosInstallTests_find_package_Trilinos](https://trilinos-cdash.sandia.gov/test/10050314) [above](https://github.com/trilinos/Trilinos/pull/10775#issuecomment-1213463411) described in detail in #10865.  I can do it for you if you say the word.  (Then someone will just need to reapprove the PR).

**Comment by jhux2** [comment]
> if you merge the branch [10865-ats2-disable-shylu](https://github.com/trilinos/Trilinos/tree/10865-ats2-disable-shylu) to this topic branch [fix-crusher-modules](https://github.com/trilinos/Trilinos/tree/fix-crusher-modules) for this PR, then that changes should fix the test failure [TrilinosInstallTests_find_package_Trilinos](https://trilinos-cdash.sandia.gov/test/10050314) [above](https://github.com/trilinos/Trilinos/pull/10775#issuecomment-1213463411) described in detail in #10865. I can do it for you if you say the word. (Then someone will just need to reapprove the PR).

@bartlettroscoe Sure, please go ahead.

**Review by bartlettroscoe — APPROVED** [review]
Reapproving after the merge of https://github.com/trilinos/Trilinos/pull/10775/commits/a550abb793c732b8664150e9b5894a5007314b0f to fix the test `TrilinosInstallTests_find_package_Trilinos`.

**Review by lucbv — APPROVED** [review]
lgtm

**Comment by GrahamBenHarper** [comment]
why is [TrilinosInstallTests_doInstall](https://trilinos-cdash.sandia.gov/testSummary.php?project=1&name=TrilinosInstallTests_doInstall&date=2022-08-31) failing? is that a spurious error?

**Comment by jhux2** [comment]
@bartlettroscoe Is the error above already being tracked internally?

**Comment by bartlettroscoe** [comment]
> why is [TrilinosInstallTests_doInstall](https://trilinos-cdash.sandia.gov/testSummary.php?project=1&name=TrilinosInstallTests_doInstall&date=2022-08-31) failing? is that a spurious error?

@GrahamBenHarper, did you look at the error message?  The [test output]() shows:

```
================================================================================

TEST_1

Install enabled and built Trilinos packages (NOTE: This test will fail if the project has **any** build errors!)

Running: "/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.17.1/bin/cmake" "--install" "/scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test" "--prefix" "/scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/install"

  Writing output to file "/scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/packages/TrilinosInstallTests/doInstall.out"

--------------------------------------------------------------------------------

NO_ECHO_OUTPUT

--------------------------------------------------------------------------------

TEST_1: Return code = 1
TEST_1: Pass criteria = Zero return code [FAILED]
TEST_1: Result = FAILED
================================================================================

TEST_2

Grep doInstall.out file produced above to see any errors

Running: "grep" "-A" "50" "CMake Error" "doInstall.out"

--------------------------------------------------------------------------------

CMake Error at /scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/packages/kokkos/core/src/cmake_install.cmake:56 (file):
  file INSTALL cannot find
  "/scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/packages/kokkos/core/src/libkokkoscore.so.13.5":
  No such file or directory.
Call Stack (most recent call first):
  /scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/packages/kokkos/core/cmake_install.cmake:42 (include)
  /scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/packages/kokkos/cmake_install.cmake:42 (include)
  /scratch/trilinos/jenkins/ascic115/workspace/Trilinos_PR_clang-10.0.0/pull_request_test/cmake_install.cmake:163 (include)



--------------------------------------------------------------------------------

TEST_2: Return code = 0
TEST_2: Pass criteria = Pass Any [PASSED]
TEST_2: Result = PASSED

================================================================================
```

and note the line:

```
Install enabled and built Trilinos packages (NOTE: This test will fail if the project has **any** build errors!)
```

So that build actually has build errors.  They are just not being reported to CDash because they are using CMake 3.17.1 and there is a problem on the client side reporting only the global `cmake --build . [other options]` (see https://github.com/trilinos/Trilinos/issues/10836#issuecomment-1230849644) and those errors are not being reported by CMake 3.17.1.

The advice of Kitware (@zackgalbreath) is to upgrade CMake on all of these platforms and see if that resolves these reporting problems (which are causing #10823 and #10836).

I will state that again, **we need to upgrade Trilinos PR testing to use more recent CMake version!** (See #10355).

**Comment by jhux2** [comment]
Hmm, I see an extra commit snuck in that's not mine.  I'll submit a new PR to see if that fixes things.

**Comment by jhux2** [comment]
Closing in favor of #10970.

**Pull request closed without merging**
